File: packages/java/src/main/java/Runner.java
import model.*;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public final class Runner {
    private final RemoteProcessClient remoteProcessClient;
    private final String token;

    public static void main(String[] args) throws IOException {
        new Runner(args.length == 3 ? args : new String[] { "127.0.0.1", "31001", "0000000000000000" }).run();
    }

    private Runner(String[] args) throws IOException {
        remoteProcessClient = new RemoteProcessClient(args[0], Integer.parseInt(args[1]));
        token = args[2];
    }

    public void run() throws IOException {
        Strategy strategy = new MyStrategy();
        Map<Integer, Action> actions = new HashMap<>();
        Game game;
        remoteProcessClient.writeToken(token);
        Rules rules = remoteProcessClient.readRules();
        while ((game = remoteProcessClient.readGame()) != null) {
            actions.clear();
            for (Robot robot : game.robots) {
                if (robot.is_teammate) {
                    Action action = new Action();
                    strategy.act(robot, rules, game, action);
                    actions.put(robot.id, action);
                }
            }
            remoteProcessClient.write(actions);
        }
    }
}

File: packages/kotlin/src/main/kotlin/Runner.kt
import model.*

import java.io.IOException
import java.util.HashMap

class Runner @Throws(IOException::class)
private constructor(args: Array<String>) {
    private val remoteProcessClient: RemoteProcessClient
    private val token: String

    init {
        remoteProcessClient = RemoteProcessClient(args[0], Integer.parseInt(args[1]))
        token = args[2]
    }

    @Throws(IOException::class)
    fun run() {
        val strategy = MyStrategy()
        val actions = HashMap<Int, Action>()
        var game: Game?
        remoteProcessClient.writeToken(token)
        val rules = remoteProcessClient.readRules()
        while (true) {
            game = remoteProcessClient.readGame()
            if (game == null) {
                break
            }
            actions.clear()
            for (robot in game.robots) {
                if (robot.is_teammate) {
                    val action = Action()
                    strategy.act(robot, rules, game, action)
                    actions[robot.id] = action
                }
            }
            remoteProcessClient.write(actions)
        }
    }

    companion object {

        @Throws(IOException::class)
        @JvmStatic
        fun main(args: Array<String>) {
            Runner(if (args.size == 3) args else arrayOf("127.0.0.1", "31001", "0000000000000000")).run()
        }
    }
}
